core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -251,6 +251,13 @@ private[spark] object JettyUtils extends Logging {
   private def attachPrefix(basePath: String, relativePath: String): String = {
     if (basePath == "") relativePath else (basePath + relativePath).stripSuffix("/")
   }
+
+  /**
+   * Attach a prefix to the given path, don't remove the trailing / if present.
+   */
+  private def attachPrefixForRedirect(basePath: String, relativePath: String): String = {
```

Comment:
Yes but I'm curious why `basePath == ""` plus `relativePath == "foo/"` should retain its trailing slash but not when `basePath` is non-empty. Is it really `basePath =="" && relativePath =="/"` that should be handled specially?

I do think it would be better to inline them. I am not sure why the suffix-stripping logic exists in the first place but that may be too far afield.